src/ipvs/ip_vs_core.c

```diff
@@ -33,18 +35,23 @@
 #include "ipvs/blklst.h"
 #include "ipvs/proto_udp.h"
 
-#define icmp4_id(icmph)      (((icmph)->un).echo.id)
-
-static inline int dp_vs_fill_iphdr(int af, const struct rte_mbuf *mbuf, 
+static inline int dp_vs_fill_iphdr(int af, struct rte_mbuf *mbuf,
                                    struct dp_vs_iphdr *iph)
 {
     if (af == AF_INET) {
-        const struct ipv4_hdr *ip4h = ip4_hdr(mbuf);
+        struct ipv4_hdr *ip4h = ip4_hdr(mbuf);
         iph->af     = AF_INET;
         iph->len    = ip4_hdrlen(mbuf);
         iph->proto  = ip4h->next_proto_id;
         iph->saddr.in.s_addr = ip4h->src_addr;
         iph->daddr.in.s_addr = ip4h->dst_addr;
+    } else if (af == AF_INET6) {
+        struct ip6_hdr *ip6h = ip6_hdr(mbuf);
+        iph->af         = AF_INET6;
+        iph->len        = ip6_skip_exthdr(mbuf, mbuf->l3_len, &ip6h->ip6_nxt);
```

Comment:
int ip6_skip_exthdr(const struct rte_mbuf *mbuf, int start, uint8_t *nexthdrp)
ip6h->ip6_nxt is just next proto or next exthdr proto name such as IPPROTO_ICMPV6, not mbuf data. Why using pointer?

Reply:
`nexthdrp` is a value-result param. fixed.